.github/workflows/release-and-bump.yaml

```diff
@@ -0,0 +1,58 @@
+name: Build GitHub release and bump SNAPSHOT
+
+on: [workflow_dispatch] # enable manual triggering of this action
+
+jobs:
+  github-release:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Get Project Version
+        id: set-release-version
+        run: |
+          current_version=$(mvn help:evaluate -f pom.xml -Dexpression=project.version -q -DforceStdout)
+          echo "Current version: $current_version"
+          new_version=$(echo $current_version | sed 's/-SNAPSHOT//')
+          echo "New version: $new_version"
+          mvn -B versions:set -f pom.xml -DnewVersion=$new_version
+          git config --global user.email "github-actions@github.com"
+          git config --global user.name "GitHub Actions"
+          git add pom.xml
+          git commit -m "Bump version to $new_version"
+          git push
+          echo "new_version=$new_version" >> $GITHUB_OUTPUT
+
+      - name: Create GitHub Release
+        id: create_release
+        uses: softprops/action-gh-release@v2
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ steps.set-release-version.outputs.new_version }}
+          name: ${{ steps.set-release-version.outputs.new_version }}
+          draft: false
+          prerelease: false
+
+  increase-snapshot:
+    needs: github-release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set Maven Project Version
+        id: bump-snapshot-version
+        run: |
+          git config --global user.email "github-actions@github.com"
+          git config --global user.name "GitHub Actions"
+          git pull
+          current_version=$(mvn help:evaluate -f pom.xml -Dexpression=project.version -q -DforceStdout)
+          echo "Current version: $current_version"
+          new_version=$(echo $current_version | awk -F. -v OFS=. '{$3=$3+1; print $0"-SNAPSHOT"}')
+          echo "New version: $new_version"
+          mvn -B versions:set -DnewVersion=$new_version -f pom.xml
```

Comment:
verstehe ich nicht, warum das so gemacht wird und nicht wie intern mit einem Maven Release. https://git.muenchen.de/ccse/cicd/gitlab-runner-templates/-/blob/main-2.x/MavenRelease.gitlab-ci.yml?ref_type=heads

Reply:
👍  für die Verwendung von maven-release-plugin

Reply:
hi, ich habe einmal das probiert.
```yml
name: maven-release
on:
  workflow_dispatch:
    inputs:
      releaseVersion:
        description: "Default version to use when preparing a release."
        required: true
        default: "X.Y.Z"
      developmentVersion:
        description: "Default version to use for new local working copy."
        required: true
        default: "X.Y.Z-SNAPSHOT"

jobs:
  release:
    runs-on: ubuntu-latest

    steps:
      # Checkout source code, set up Java, etc. Then...
      - name: Checkout code
        uses: actions/checkout@v4
      - name: Set up JDK 17
        uses: actions/setup-java@v4
        with:
          java-version: "17"
          distribution: "temurin"
          cache: "maven"
      - name: Maven Release
        run: |
           git config --global user.email "github-actions@github.com"
           git config --global user.name "GitHub Actions"
           mvn release:prepare release:perform -B  -Dpassword=${GITLAB_ACCESS_TOKEN_PW}  -DreleaseVersion=${{ github.event.inputs.releaseVersion }} -DdevelopmentVersion=${{ github.event.inputs.developmentVersion }}
        env:
           GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

`fatal: could not read Username for 'https://github.com/': No such device or address`

also das pushen und taggen hat alles funktioniert. nur kommt ein fehler. intern machen wir ja nur ein `release:prepare`. das hat auch glaube ich mit dem token etwas zu tun.

https://github.com/it-at-m/digiwf-core/blob/dev/.github/workflows/workflow-build-services.yaml#L90 digi wf nutzt einen selbst erstellten Service-Token